### PR TITLE
Wallet vue adapter

### PR DESCRIPTION
**Pull request by ildarH**
A new wallet adapter for Vue 3 that can be easily integrated with both Vue 3 and Nuxt 3 applications.

This PR adds wallet adapter and example for Nuxt 3

### CODE REVIEW

**Review by 0xmaayan — COMMENTED** [review]
Thank you for this amazing contribution! I tried the demo vue dapp locally, but could run it only when I removed mSafe wallet from the [wallets list](https://github.com/aptos-labs/aptos-wallet-adapter/pull/363/files#diff-5b598e2a2febe5ee3f4e0d4322f2e247d4e252d165c034d564befd0e90fd3d65R22) - any idea why?

**Comment by 0xmaayan** [comment]
> Thank you for this amazing contribution! I tried the demo vue dapp locally, but could run it only when I removed mSafe wallet from the [wallets list](https://github.com/aptos-labs/aptos-wallet-adapter/pull/363/files#diff-5b598e2a2febe5ee3f4e0d4322f2e247d4e252d165c034d564befd0e90fd3d65R22) - any idea why?

Ok, it looks like it is because `@msafe/aptos-wallet-adapter` is not part of the dapp dependencies array. Actually non of the wallet packages are in the dependencies array - why?

**Comment by ildarH** [comment]
> > Thank you for this amazing contribution! I tried the demo vue dapp locally, but could run it only when I removed mSafe wallet from the [wallets list](https://github.com/aptos-labs/aptos-wallet-adapter/pull/363/files#diff-5b598e2a2febe5ee3f4e0d4322f2e247d4e252d165c034d564befd0e90fd3d65R22) - any idea why?
> 
> Ok, it looks like it is because `@msafe/aptos-wallet-adapter` is not part of the dapp dependencies array. Actually non of the wallet packages are in the dependencies array - why?

Hey! I've updated. Sorry for the misunderstanding. Can you check please?

**Comment by 0xmaayan** [comment]
Overall LGTM. Thank you for this contribution!

Some notes:

1. We are missing the transaction hash in the link after successfully submitting a transaction (attached screenshot)
2. Is there a reason you didn't include `Antd` and `MUI` wallet selectors in the demo app?
3. When the wallet network is set to `Mainnet` we should show a warnning message as we have on the nextjs demo app
4. Given this is a community contribution, and to provide the best support, we should lock the [@aptos-labs/wallet-adapter-core version](https://github.com/aptos-labs/aptos-wallet-adapter/pull/363/files#diff-d6f882034fc991cc9d912f2fa1ab77e82ad7c849f6e31818eb0a5b92c93e473aR44) to the latest version this Provider supports. This is to ensure we dont break anything in this package when we do `core` package upgrades 
 
image

**Comment by ildarH** [comment]
Thank you for your feedback! I've added fixes.
Regarding the second point, I can include some of the libraries, such as Antd (it has vue version for sure) a bit later in a separate pull request.

File: packages/wallet-adapter-vue/package.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    "Nuxt"
+  ],
+  "scripts": {
+    "build:bundle": "vite build",
+    "build:declarations": "vite build",
+    "types": "vue-tsc",
+    "build": "pnpm build:bundle"
+  },
+  "dependencies": {
+    "@aptos-labs/wallet-adapter-core": "^4.15.0"
```
Comment (0xmaayan) [inline packages/wallet-adapter-vue/package.json]: lets lock it to a specific version
```suggestion
    "@aptos-labs/wallet-adapter-core": "4.15.0"
```

**Review by 0xmaayan — APPROVED** [review]
LGTM!

**Comment by 0xmaayan** [comment]
@ildarH build is failing, could you run `pnpm install` on the adapter root folder, and then make sure build succeed with `pnpm turbo run build`?
https://github.com/aptos-labs/aptos-wallet-adapter/actions/runs/10563236480/job/29324042334?pr=363

**Pull request merged**